src/main/scala/viper/gobra/frontend/info/implementation/typing/ExprTyping.scala

```diff
@@ -431,9 +431,15 @@ trait ExprTyping extends BaseTyping { this: TypeInfoImpl =>
         case PFPredBase(id) =>
           val idT = idType(id)
           PredT(idT.asInstanceOf[FunctionT].args.zip(args).collect{ case (typ, None) => typ })
-        case p: PMPredBase =>
-          val idT = idType(p.id)
-          PredT(idT.asInstanceOf[FunctionT].args.tail.zip(args).collect{ case (typ, None) => typ })
+        case p: PDottedBase => resolve(p.recvWithId) match {
+          case Some(_: ap.Predicate) =>
+            val recvWithIdT = exprOrTypeType(p.recvWithId)
+            PredT(recvWithIdT.asInstanceOf[FunctionT].args.zip(args).collect{ case (typ, None) => typ })
+          case Some(_: ap.PredicateExpr) => ??? // Predicate expressions are not supported at the moment
```

Comment:
Currently, we also do not plan to support them in the future. With my latest proposal, they do not make much sense.

Reply:
I will change the type system to complain when a Predicate Expression is used in a predicate constructor

Reply:
@Felalolf Why do `PredicateExpr` not make much sense with your latest proposal?
For me, `PredicateExpr` and `ReceivedPredicate` are two different forms for the equivalent thing (like Go does it for methods)

Reply:
I mixed up predicate method expressions and values with a predicate type. Former is fine, later is not.

Reply:
Makes sense